lib/spack/llnl/util/filesystem.py

```diff
@@ -424,6 +424,19 @@ def fix_darwin_install_name(path):
                     break
 
 
+def to_link_flags(library):
+    """Transforms a path to a <library> into linking flags -L<dir> -l<name>.
+
+    Return:
+      A string of linking flags.
+    """
+    dir  = os.path.dirname(library)
+    # Asume   libXYZ.suffix
+    name = os.path.basename(library)[3:].split(".")[0]
+    res = '-L%s -l%s' % (dir,name)
```

Comment:
my understanding that it is portable across compiler/linkers. One could extend it with extra options, but for now that's all I need in `superlu-dist`.